### PR TITLE
Adding The tale of Tamiyo (v1.0)

### PR DESCRIPTION
**Pull request by frafen**
EDIT: I should mention that I made this card with the help of Chat GPT and that I have no coding experience. My process was to base the first ability of the saga on Grindstone, and the second on Mizzix's Mastery's overload ability. Than I used chat GPT to fix the errors that InteljIDEA signaled to me. Finally I tested the card.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+    private final FilterCard filter;
+
+    TheTaleOfTamiyoEffect2() {
+        super(Outcome.PlayForFree);
+        this.staticText = "Exile any number of target instant, sorcery, and/or Tamiyo planeswalker cards " +
+                "from your graveyard. Copy them. You may cast any number of the copies.";
+
+        // Initialize filter
+        this.filter = new FilterCard("instant, sorcery, or Tamiyo planeswalker card");
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java:143]: make the filter `static` and put this initialization in a `static` block after it, not in the constructor.

File: Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                if (controller.moveCards(cardsToExile, Zone.EXILED, source, game)) {
+                    Cards copiedCards = new CardsImpl();
+                    // Copy cards
+                    for (Card card : cardsToExile) {
+                        copiedCards.add(game.copyCard(card, source, source.getControllerId()));
+                    }
+
+                    // Cast copies
+                    boolean continueCasting = true;
+                    while (controller.canRespond() && continueCasting && !copiedCards.isEmpty()) {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java:179]: Do not use a while loop for this. Instead, use Integer.MAX_VALUE for the max number of targets in the target card, so all the cards to cast can be selected at once. Then for the casting process iterate through those targets with a for loop.

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+            // Stop if there are not enough cards
+            if (controller.getLibrary().size() < 2) {
+                return true;
+            }
+
+            do {
+                iteration++;
+                if (iteration > possibleIterations + 20) {
+                    // Protection against infinity loops
```
[the remark this answers is not in the document]
Comment (frafen) [reply Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java:87]: I don't know either, I copy/pasted it from Grindstone's code (that is the base of this effect's code)

File: Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+                }
+
+                // If there is a match, draw a card and continue; otherwise, break
+                if (typesMatch) {
+                    controller.drawCards(1, source, game);
+                } else {
+                    break;
+                }
+
+            } while (controller.getLibrary().size() >= 2); // Continue if there are at least 2 cards left in the library
```
[the remark this answers is not in the document]
Comment (frafen) [reply Mage.Sets/src/mage/cards/t/TheTaleOfTamiyo.java:123]: I think that I can fix this. For the other requests I'm afraid I can't do much, my coding experience is almost nonexistant. I tried to fix some of those other problems with the help of Chat gpt but I'm not having much success.

**Pull request closed without merging**